### ISSUE TITLE
ElasticPress_CLI_Command::stop_the_insanity() breaks WooCommerce

### ISSUE THREAD

**Issue by dnaber-de** [10up/ElasticPress#691]
Hello it's me again,

I discovered an issue in combination with WooCommerce and it's shortcodes while indexing via WP_CLI:

```
$ wp elasticpress index --setup

Fatal error: Uncaught Error: Call to a member function is_downloadable() on boolean in /.../wp-content/plugins/woocommerce/includes/wc-template-functions.php:701
Stack trace:
#0 /.../wp-content/themes/my-theme/woocommerce/content-single-product.php(44): woocommerce_get_product_schema()
#1 /.../wp-includes/template.php(574): require('/var/www/projec...')
#2 /.../wp-content/plugins/woocommerce/includes/wc-core-functions.php(173): load_template('/var/www/projec...', false)
#3 /.../wp-content/plugins/woocommerce/includes/class-wc-shortcodes.php(726): wc_get_template_part('content', 'single-product')
#4 /.../wp-includes/shortcodes.php(326): WC_Shortcodes::product_page(Array, '', 'product_page')
#5 [internal function]: do_shortcode_tag(Array)
...
```

The actual trigger is a post that uses the `product_page` shortcode which performs a WP_Query and `WP_Query::the_post()` which triggers the action `the_post` which triggers the listener `wc_setup_product_data()` which calls the function `wc_get_product()` which asks for `did_action( 'woocommerce_init' );` and leaves the global `$product` to be `false` in case `did_action()` returns false. That in turn causes almost any WooCommerce template function that relies on this global variable to crash.

However, reseting the `$wp_action` global in `ElasticPress_CLI_Command::stop_the_insanity()` might not be the best idea from this perspective as it will break any `did_action()` check out there.

**Comment by tlovett1** [comment]
Testing this with ElasticPress 2.2 and WC 2.6.13. Not able to reproduce in WP-CLI sync.

**Comment by dnaber-de** [comment]
Just for the record: solved by 1417fb14 but only for >=2.2 which introduce breaking changes in comparison to 2.1.
